lib/value-pairs/tests/test_value_pairs.c

```diff
@@ -122,85 +127,74 @@ testcase(const gchar *scope, const gchar *exclude, const gchar *expected, GPtrAr
   value_pairs_foreach(vp, vp_keys_foreach, msg, 11, LTZ_LOCAL, &template_options, args);
   g_list_foreach(vp_keys_list, (GFunc) cat_keys_foreach, vp_keys);
 
-  if (strcmp(vp_keys->str, expected) != 0)
-    {
-      fprintf(stderr, "Scope keys mismatch, scope=[%s], exclude=[%s], value=[%s], expected=[%s]\n", scope,
-              exclude ? exclude : "(none)", vp_keys->str, expected);
-      success = FALSE;
-    }
+  cr_expect_str_eq(vp_keys->str, expected, "Scope keys mismatch, scope=[%s], exclude=[%s], value=[%s], expected=[%s]",
+                   scope, exclude ? exclude : "(none)", vp_keys->str, expected);
+  cr_expect(test_key_found, "test.key is not found in the result set");
 
-  if (!test_key_found)
-    {
-      fprintf(stderr, "test.key is not found in the result set\n");
-      success = FALSE;
-    }
   g_list_foreach(vp_keys_list, (GFunc) g_free, NULL);
   g_list_free(vp_keys_list);
   g_string_free(vp_keys, TRUE);
   log_msg_unref(msg);
   value_pairs_unref(vp);
 }
 
-int
-main(int argc, char *argv[])
+struct value_pairs_params
 {
+  gchar *scope;
+  gchar *exclude;
+  gchar *expected;
   GPtrArray *transformers;
-  GlobalConfig *cfg;
+};
 
-  app_startup();
-  putenv("TZ=MET-1METDST");
-  tzset();
+ParameterizedTestParameters(value_pairs, test)
+{
+  const static struct value_pairs_params params[] =
```

Comment:
I like the `const` here, I didn't know that `cr_make_param_array` accepts it (Well I checked, it's just a macro that cast the array to a void pointer 😄 ).

Although, you didn't use `const` in `test_strtok_r.c`.